rpcserver.go

```diff
@@ -1522,6 +1686,13 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 // returned.
 func (r *rpcServer) LookupInvoice(ctx context.Context,
 	req *lnrpc.PaymentHash) (*lnrpc.Invoice, error) {
+	// Check macaroon to see if this is allowed.
+	if r.server.authSvc != nil {
+		if err := macaroons.ValidateMacaroon(ctx, "lookupinvoice",
```

Comment:
On the topic of using the same permission for the same data, should `lookupinvoice` be the same permission as `listinvoices`? The reason I chose not to do this is that `lookupinvoice` requires a known payment hash and retrieves a single invoice, whereas `listinvoices` allows the caller to retrieve **all** invoices indiscriminately and also to subscribe to invoice notifications.